app/models/user.rb

```diff
@@ -1,8 +1,40 @@
 # frozen_string_literal: true
 
+require 'pdf-reader'
+
 class User < ActiveRecord::Base
   validates_presence_of :first_name, :last_name, :email
   validates_uniqueness_of :email
 
   has_many :residents, -> { order(position: :asc) }
+
+  def load_residents(file)
+    reader = PDF::Reader.new(file[:tempfile].path)
+    save_next_line = false
+
+    reader.pages.each do |page|
+      page.text.each_line do |line|
+        # Remove blank lines
+        next if line.strip.empty?
+
+        if save_next_line
```

Comment:
Without seeing an example of this PDF it's tough to say for sure, but this seems really complicated. There must be some data interspersed with the names, right? So we're basically using information from the header to determine if we save the next line, which should be a name?

If so, then we kindof have a Data Clump. https://refactoring.guru/smells/data-clumps

If we apply one of the treatments here and combine the "header"  and the "data" into one object, we could simplify this logic a bit. That will make it easier to reason about and less error-prone as a result.

Reply:
Yeah there's probably a much better way to do this. The names are all on their own line. Below each line is link to pray, care, share, etc. So I'm basically looking for those link names and taking the line after it (after blank lines are removed). And then I have to look for the first line as well so I can get the first name. Maybe we could spend a little time on this over the weekend if you still have some time to hang.

Reply:
Ok, so maybe we can filter out the links and just process a list of names one by one?